src/components/UserIcon/UserIcon.js

```diff
@@ -44,7 +43,7 @@ export default function User() {
     };
   }, []);
 
-  const bakery = useSelector(getBakery);
+  const getState = useStore().getState;
```

Comment:
Unfortunately react hooks can enforce poor coding practices like this because they must be always called in the same order so they cannot be put in callbacks or conditionals. Yes this means that the majority of the time they do work that's not necessary so we try to memoize their responses at least. In this case it's simply calling a function that returns a function to get the state so it should have minimal performance implications 🤷‍♂